docs/user_guide/usage_patterns/data_transfer/detecting_data_access/index.rst

```diff
@@ -0,0 +1,161 @@
+.. currentmodule:: globus_sdk
+
+.. _userguide_detecting_data_access:
+
+Detecting data_access
+=====================
+
+Globus Collections come in several varieties, but only some of them have a
+``data_access`` scope.
+
+``data_access`` scopes control application access to collections, allowing
+users to revoke access for an application independent from other application
+permissions.
+Revoking consent stops data transfers and other operations.
+
+Because only some collection types have ``data_access`` scopes, application
+authors interacting with these collections may need to detect the type of
+collection and determine whether or not the scope will be needed.
+
+For readers who prefer to start with complete working examples, jump ahead to the
+:ref:`example script <userguide_detecting_data_access_example>`.
+
+Accessing Collections in Globus Transfer
+----------------------------------------
+
+The Globus Transfer service acts as a central registration hub for collections.
+Therefore, in order to get information about an unknown collection, we will
+need a :class:`TransferClient` with credentials.
+
+The following snippet creates a client and uses it to fetch a collection from
+the service:
+
+.. code-block:: python
+
+    import globus_sdk
+
+    # Tutorial Client ID - <replace this with your own client>
+    NATIVE_CLIENT_ID = "61338d24-54d5-408f-a10d-66c06b59f6d2"
+    USER_APP = globus_sdk.UserApp("detect-data-access-example", client_id=NATIVE_CLIENT_ID)
+
+    # Globus Tutorial Collection 1
+    # https://app.globus.org/file-manager/collections/6c54cade-bde5-45c1-bdea-f4bd71dba2cc
+    # replace with your own COLLECTION_ID
+    COLLECTION_ID = "6c54cade-bde5-45c1-bdea-f4bd71dba2cc"
+
+    transfer_client = globus_sdk.TransferClient(app=USER_APP)
+
+    collection_doc = transfer_client.get_endpoint(COLLECTION_ID)
+
+.. caution::
+
+    Careful readers may note that we the :meth:`TransferClient.get_endpoint`
+    method to lookup a collection.
+
+    The Transfer service contains both Endpoints and Collections, and for
+    historical reasons both document types are available from the Get Endpoint
+    API.
+    This is the correct API to use not only when looking for information about
+    a Collection, but also for detecting whether an ID belongs to a Collection
+    or Endpoint, in cases where it is ambiguous.
```

Comment:
Are these intended to be separate paragraphs?

Separately, it might not be worthwhile to mention "for historical reasons", but rather to say simply that "this API supports both collections and endpoints" (but I'm not convinced that there's a strong need to even have this caution block).

Reply:
Not separate paragraphs, but a linebreak after the period to make them more readable as raw text (maybe poor style? we don't have a strong convention around this as of yet).

I'd rather not nix the whole block, although I understand and agree that removing it wouldn't be the _worst_. I think some users might notice that we've told them to use `get_endpoint` to lookup a **collection** and think we've made a mistake in our docs. Maybe I should tone it down to a `note` block?

I'll look at rephrasing it, and maybe reflow it when I do if you have any thoughts on the linebreak style.

Reply:
I generally err toward breaks at conjunctions and prepositions, like:

```suggestion
    The Transfer service contains both Endpoints and Collections,
    and for historical reasons both document types are available
    from the Get Endpoint API.
    This is the correct API to use not only when looking for information about
    a Collection, but also for detecting whether an ID belongs to a Collection
    or Endpoint, in cases where it is ambiguous.
```

Downgrading from `caution` to `note` will alleviate some of my concerns here. :+1:

Reply:
I made it a `note`, folded at the `"and"`, and also cut the last sentence as unnecessary / too long. :tada: